lib/beaker/dsl/install_utils/foss_utils.rb

```diff
@@ -689,23 +689,38 @@ def install_puppet_from_dmg_on( hosts, opts )
         # @return nil
         # @api private
         def install_puppet_agent_from_dmg_on(hosts, opts)
+          require 'nokogiri'
```

Comment:
The code referenced above doesn't specify that it needs to be the "latest", or use the latest idea that we've developed for PE. You should be able to point at whatever file resource you'd like at any URL, it just matters that the file only contain the version that you'd like to use for that beaker run.

You wouldn't want to use that code anyway because it's for PE, and we don't want FOSS code depending on it. You would just be copying the pattern of how versions are scraped from there.  If you copy it over in a nice general method, then we can actually just make beaker-pe use your method, leaving that method as an adapter if needed.

If this is unclear, maybe we should just get together for a quick face-to-face to get on the same page about what's needed here & what would be the best way to do it.